addons/web_editor/static/src/js/editor/snippets.options.js

```diff
@@ -4191,7 +4203,7 @@ registry.BackgroundShape = SnippetOptionWidget.extend({
         }
         const searchParams = Object.entries(colors)
             .map(([colorName, colorValue]) => {
-                const encodedCol = encodeURIComponent(colorValue);
+                const encodedCol = encodeURIComponent(normalizeColor(colorValue));
```

Comment:
@bvr-odoo Can you split your commit in two commits (same PR) as they are different bug and fixes? In fact for the second one here, I am not sure: are not we supposed to have `c4=o-color1` in the URL? Or maybe it's only from 15.0 (16.0?) and above that we allowed that to work? In any case: it's why it is really better to split in two commits, the second one is at least not needed in master. You could actually make two PR instead of two commits to potentially ease the forward-port process.

Reply:
@qsm-odoo here is the second PR https://github.com/odoo/odoo/pull/114684

In version 15 and above, no colors are defined in the `data-oe-shape-data` or on URL's of templates. After changing the "shape" options," the colors defined in `data-oe-shape-data` and in the URL are always in hexadecimal format. There are indeed colors defined like this `c4=o-color1` in URLs in the code, but only for the shapes of images or SVG illustrations.

Reply:
(note that I did not asked for two PR but two commits but in the end, it's good to have two PR since the other one did not need any forward-port :pray:)

Reply:
> You could actually make two PR instead of two commits to potentially ease the forward-port process.

?